### PR TITLE
FB-1038 Move stat logging to ES client, log index names and types

### PR DESCRIPTION
**Pull request by tomaszdurka**
Reverts cargomedia/cm#2677

### CODE REVIEW

File: library/CM/Elasticsearch/Factory.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        $client = $this->createClient($servers, $debug);
+        return new CM_Elasticsearch_Cluster($client, $disabled);
+    }
+
+    /**
+     * @param array         $servers
+     * @param CM_Debug|null $debug
+     * @return CM_Elasticsearch_Client
+     */
+    public function createClient(array $servers, $debug = null) {
```
Comment (fvovan) [inline library/CM/Elasticsearch/Factory.php]: Please add type hinting for `$debug`

File: library/CM/Elasticsearch/Client.php [hunk 2]
```diff
@@ -264,12 +269,16 @@ public function search($indexNameList, $typeNameList, array $data) {
             'type'  => self::_prepareIndexNameParam($typeNameList),
             'body'  => $data,
         ];
+
+        if ($debug = $this->_getDebug()) {
+            $debug->incStats('search', json_encode($params));
```
Comment (fvovan) [inline library/CM/Elasticsearch/Client.php]: I think it's better to use CM_Utils::jsonEncode()

**Comment by fvovan** [comment]
rest looks good, the only thing came to my mind is  probably we don't even need `Cluster` class now.
It's not a container anymore and all it does is just proxifying queries to `Client` (which is kind of proxy too). well also it allows to enable/disable cluster..

**Comment by tomaszdurka** [comment]
Yes, but I didn't want to refactor it here. Or maybe.

**Comment by tomaszdurka** [comment]
I will keep it, it has this "enabled" feature which we should keep for now.

**Pull request merged**